src/app/core/services/courses.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { Course } from "@core/models/course";
+import { CoursesData } from "@mock/courses";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CoursesService {
+
+  constructor() { }
+
+  getCourses(): Course[] {
+    return CoursesData;
+  }
+
+  addCourse(newCourse: Course) {
+    console.log('new course was added', newCourse);
+  }
+
+  updateCourse(id: number, newCourseData: Course) {
+    console.log('update course', newCourseData);
+  }
+
+  deleteCourse(id: number) {
```

Comment:
Logic to open a dialog should not be in service. And it should not be a confirm method